### PR TITLE
Add short/long-path beam heading for directional-antenna operators

### PR DESCRIPTION
**Pull request by patrickrb**
## What & why

Operators running a beam — or any directional antenna — need to know **which way to point it** to work a station. The QSO sheet already showed a single short-path azimuth, but nothing surfaced the **long-path** heading (the reciprocal, used when the short path is closed), and you had to open a station's sheet to see a heading at all.

This PR makes beam heading first-class:

- **QSO sheet** — the *Azimuth* stat card now leads with the short-path heading and shows the long-path heading as a subtitle (`LP 227°`), so DXers can see both ways to aim the antenna at a glance.
- **Decode list (opt-in)** — a compact heading column next to distance, so you can aim before you even tap a station. Gated by a new **Show beam heading** toggle (Settings → Decode filters), **off by default** so vertical/wire operators don't get a column they don't need.

## How

New pure helper `ui/decode/BeamHeading.kt`:

- `greatCircleBearing` — standard forward-azimuth (initial great-circle bearing).
- `longPathBearing` — reciprocal (`short + 180`, wrapped).
- `normalizeHeadingDeg` — round to whole degrees in `0..359` (360 → 0).
- `computeBeamHeadings(myGrid, theirGrid)` — decodes both grids via `MaidenheadGrid`; returns `null` for missing / unparseable / identical grids (a bearing to yourself is undefined).

The trig is Android-free so it unit-tests without Robolectric. The QSO sheet and the decode row both go through `computeBeamHeadings`, so the two surfaces can never disagree. The old private, short-path-only azimuth helper in `QsoSheet.kt` is replaced.

Config: new `GeneralVariables.showBeamHeading` (persisted as `showBeamHeading`, loaded in `DatabaseOpr` next to `distanceInMiles`).

## Tests

- `BeamHeadingTest` — pure math: cardinal bearings (E/W/N/S), `0..360` normalisation, long-path reciprocal + wrap, rounding edge cases (`359.7 → 0`), formatting.
- `BeamHeadingGridTest` (Robolectric) — grid decoding: null cases, long-path = reciprocal, range bounds, Boston→London reads NE (~50°), due-east reads ~90°, plus the decode-row wrapper `computeBeamHeadingText`.

## Verification

- `./gradlew :app:testDebugUnitTest --tests 'radio.ks3ckc.ft8af.ui.decode.*'` — green.
- `./gradlew :app:assembleDebug` — builds successfully (all 4 ABIs).

## Compatibility

Display-only; no change to the FT8/WSJT-X protocol, decoder, or TX path. The decode-row addition is opt-in and default-off, so existing users see no change until they enable it.

### CODE REVIEW

File: ft8af/app/src/main/kotlin/radio/ks3ckc/ft8af/ui/decode/BeamHeading.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * The trig lives in [greatCircleBearing]/[longPathBearing] as pure functions (no
+ * Android types) so they unit-test without Robolectric; [computeBeamHeadings]
+ * decodes the grids through [MaidenheadGrid] and therefore needs it.
+ */
+internal data class BeamHeadings(val shortPathDeg: Int, val longPathDeg: Int)
+
+/**
+ * Initial great-circle bearing (degrees, `0..360`) from point 1 to point 2 using
+ * the standard forward-azimuth formula. Pure math — no Android types.
+ */
```
[the remark this answers is not in the document]
Comment (patrickrb) [reply ft8af/app/src/main/kotlin/radio/ks3ckc/ft8af/ui/decode/BeamHeading.kt:27]: Fixed. The KDoc now documents `[0, 360)` and notes that the final `% 360.0` means 360 is never returned (a due-north bearing is 0).
